iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -215,6 +219,158 @@ void PortManager::doDiscoveryForSubscriberPort(SubscriberPortType& subscriberPor
     });
 }
 
+void PortManager::destroyClientPort(popo::ClientPortRouDi::MemberType_t* const clientPortData) noexcept
+{
+    cxx::Ensures(clientPortData != nullptr && "clientPortData must not be a nullptr");
+
+    // create temporary client ports to orderly shut this client down
+    popo::ClientPortRouDi clientPortRoudi(*clientPortData);
+    popo::ClientPortUser clientPortUser(*clientPortData);
+
+    clientPortRoudi.releaseAllChunks();
+    clientPortUser.disconnect();
+
+    // process DISCONNECT for this client in RouDi and distribute it
+    clientPortRoudi.tryGetCaProMessage().and_then([this, &clientPortRoudi](auto caproMessage) {
+        cxx::Ensures(caproMessage.m_type == capro::CaproMessageType::DISCONNECT);
+
+        m_portIntrospection.reportMessage(caproMessage);
+        this->sendToAllMatchingServerPorts(caproMessage, clientPortRoudi);
+    });
+
+    /// @todo iox-#27 report to port introspection
+
+    // delete client port from list after DISCONNECT was processed
+    m_portPool->removeClientPort(clientPortData);
+
+    LogDebug() << "Destroyed client port";
```

Comment:
I would think it maybe helpful to add the process name and service description here.

Reply:
Unfortunately the process name is not available at this point, but I added the service description